crates/sparrow-main/src/serve/compute_service.rs

```diff
@@ -221,6 +277,27 @@ async fn execute_impl(
         .map(|item| item.into_status()))
 }
 
+fn materialize_impl(
```

Comment:
This should be renamed -- it doesn't actually do the materializing, it just creates one.
It looks like the reason the materializations are in an `Arc` is so they can be passed here, but that is only used for inserting in the map. I'd suggest instead having this return the `control_handle` and then having the caller insert into the materializations. This will in turn mean we don't need to clone the materializations to pass them in here.

Also, not sure splitting this method out at all benefits that much, but your call.

Reply:
Done.

The benefit of splitting it into this method gives me one place to handle the error, which I like.